design-proposals/edge-infra-day2-improvements.md

```diff
@@ -0,0 +1,485 @@
+    end
+
+    loop Edge Node polling
+        PUA->>MM: PlatformUpdateStatusRequest
+        note left of PUA: Response contains information from update_policy and the scheduled update
+        MM->>PUA: PlatformUpdateStatusResponse
+    end
+    
+    note over MM,PUA: Update Schedule Start
+    PUA->>PUA: Start Update
+    PUA->>MM: PlatformUpdateStatusRequest with updateStatu "STATUS_TYPE_STARTED"
+    MM->>Inventory: Create a OSUpdateRun, with start_time, linking to the OSUpdatePolicy
+    
+    alt Update successful on the Edge Node
+        Edge Node->>MM: PlatformUpdateStatusRequest with updateStatus "STATUS_TYPE_UPDATED"
+        MM->>Inventory: Update OSUpdateRun with the IDLE status and end_time
+    else Update failed on the Edge Node
+        Edge Node->>MM: PlatformUpdateStatusRequest with updateStatus "STATUS_TYPE_FAILED"
+        MM->>Inventory: Update OSUpdateRun with the ERROR status, end_time and status_details
+    end
+```
+
+Automatic Update of the Instance `os_updates_available` field:
+```mermaid
+sequenceDiagram
+%%{wrap}%%
+autonumber
+    box Edge Infrastructure Manager
+    participant Inventory
+    participant MM as Maintenance Manager
+    end
+    box Edge Node
+    participant PUA as Platform Update Agent
+    end
+    
+    loop Peridic calls from EN to MM
+    alt Immutable (EMT) OS
+    PUA->>MM: PlatformUpdateRequest
+        note over MM,Inventory: New EMT version is detected by Maintenance Manager
+        MM->>Inventory: Update "os_updates_available" field on the Instance with the new EMT version
+        MM->>PUA: PlatformUpdateResponse
+    else Mutable (Ubuntu) OS
+        note over MM,PUA: New package updates are detected by PUA
+        PUA->>MM: PlatformUpdateRequest with new package versions
+        MM->>Inventory: Update "os_updates_available" field on the Instance with the list of new package versions available
+        MM->>PUA: PlatformUpdateResponse
+    end
+    end
+```
+
+Installed packages filled in OSProfile and runtime packages in Instances:
+```mermaid
+sequenceDiagram
+%%{wrap}%%
+autonumber
+    box Edge Infrastructure Manager
+    participant Inventory
+    participant OSRM as OS Resource Manager
+    participant uRepo as Ubuntu repository
+    participant MM as Maintenance Manager
+    end
+    box Edge Node
+    participant PUA as Platform Update Agent
+    end
+    
+    
+    loop OSRM creates the OSProfile for Mutable OS (Ubuntu)
+        OSRM->>uRepo: Fetch the Manifest from the Ubuntu repository
+        OSRM->>OSRM: Parses the Ubuntu Manifest file
+        OSRM->>Inventory: Packages info into the "installed_packages" field in the OSProfile
+    end
+    
+    loop Edge Node checking for installed packages on the OS
+        MM->>PUA: List of installed packages
+        PUA->>Inventory: Compare and update "runtime_packages" field in the Instance if needed
+    end
+```
+
+### Orchestrator Update and Migrations
```

Comment:
I am just worried about the effort we are going to ask UI

Reply:
I know, I've discussed with UI today. Also, we are proposing a phased approach to support this.